pkgs/os-specific/linux/raspberrypi-eeprom/default.nix

```
@@ -24,18 +24,18 @@ stdenvNoCC.mkDerivation rec {
   '';
 
   installPhase = ''
-    mkdir -p $out/bin $out/share/rpi-eeprom
+    mkdir -p "$out/bin" "$out/lib/firmware/raspberrypi/bootloader-"{2711,2712}
 
-    cp rpi-eeprom-config rpi-eeprom-update rpi-eeprom-digest $out/bin
-    cp -r firmware/{beta,critical,old,stable} $out/share/rpi-eeprom
-    cp -P firmware/default firmware/latest $out/share/rpi-eeprom
+    cp rpi-eeprom-config rpi-eeprom-update rpi-eeprom-digest "$out/bin"
+    cp -rP firmware-2711/{beta,critical,default,latest,stable,old} "$out/lib/firmware/raspberrypi/bootloader-2711"
+    cp -rP firmware-2712/{beta,critical,default,latest,stable}     "$out/lib/firmware/raspberrypi/bootloader-2712"
```

```suggestion
    cp -rP firmware-* "$out/lib/firmware/raspberrypi/"
```

---

This was intended to exclude the file [release-notes.md](https://github.com/raspberrypi/rpi-eeprom/blob/7e22b8ff470c813d0331f8ef64b6c08125b179cc/firmware-2711/release-notes.md) but I guess it's fine to copy it as well.

---

Hmm, no, this doesn't work because `firmware-2711` needs to be renamed to `bootloader-2711` and `firmware-2712` to `bootloader-2712`. I could change `FIRMWARE_ROOT` to use the prefix `firmware` instead of `bootloader` but I think we should use `bootloader` because upstream also uses this path: https://github.com/raspberrypi/rpi-eeprom/blob/v2023.10.30-2712/rpi-eeprom-update#L17.

---

Then how about?
```suggestion
    mkdir -p $out/lib/firmware/raspberrypi
    for dirname in firmware-*; do
        dirname_suffix="${dirname/#firmware-}"
        cp -rP "$dirname" "$out/lib/firmware/raspberrypi/bootloader-$dirname_suffix"
    done
```
This uses some arcane [shell parameter expansion](https://stackoverflow.com/a/52233591)

---

I had to adapt the suggestion very slightly but then it works. Thanks!